src/GitHub.VisualStudio/Services/UsageTracker.cs

```diff
@@ -145,8 +135,29 @@ public void IncrementLoginCount()
             SaveUsage(usage);
         }
 
+        void Initialize()
+        {
+            // The services needed by the usage tracker are loaded when they are first needed to
+            // improve the startup time of the extension.
+            if (userSettings == null)
+            {
+                client = uiProvider.GetService<IMetricsService>();
+                connectionManager = uiProvider.GetService<IConnectionManager>();
+                userSettings = uiProvider.GetService<IPackageSettings>();
+                vsservices = uiProvider.GetService<IVSServices>();
```

Comment:
Right, the timer is a `DispatcherTimer` so the tick method will always call this from the main thread so that's fine. I don't think anything else will be calling this from background threads, but in case that happens, probably best to make sure we're on the right thread here.